code/modules/reagents/reagent_containers/glass_containers.dm

```diff
@@ -307,6 +307,11 @@
 	container_type = OPENCONTAINER
 	dog_fashion = /datum/dog_fashion/head/bucket
 
+/obj/item/reagent_containers/glass/bucket/update_overlays()
+	. = ..()
+	if(!is_open_container())
+		. += "lid_[initial(icon_state)]"
```

Comment:
Isn't this already handled in `glass_containers.dm`, line 166 to 167?

Reply:
It is but only for beakers